tools/lg_tools.py

```
@@ -273,16 +267,130 @@ def _extract_text_from_html(content: bytes) -> str:
+        than the model's own training data ("what's my coffee order?",
+        "remind me what we decided about the auth migration").
+
+        Returns ``"No matches."`` when the store is empty or nothing
+        scores above the keyword threshold.
+        """
+        results = knowledge_store.search(query, k=k)
+        if not results:
+            return "No matches."
+        lines = []
+        for r in results:
+            lines.append(f"[{r.get('domain', '?')}] {r['preview']}")
```

Ruff (0.15.12)

[warning] Use a list comprehension to create a transformed list

(PERF401)